language/move-compiler/src/to_bytecode/optimize/mod.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::parser::ast::FunctionName;
+use move_ir_types::ast::{self as IR};
+use std::collections::{BTreeSet, HashMap};
+
+mod remove_fallthrough_jumps;
+mod remove_unnecessary_store;
+mod remove_unused_locals;
+
+pub(crate) fn code(
+    _f: &FunctionName,
+    loop_heads: &BTreeSet<IR::BlockLabel_>,
+    locals: &mut Vec<(IR::Var, IR::Type)>,
+    blocks: &mut IR::BytecodeBlocks,
+) {
```

Comment:
I wonder whether optimizations should be flag enabled. They are in solc where they play a real important part. We can also look at this later but want to bring this up already now.